code/obj/mining_cloud_storage.dm

```diff
@@ -0,0 +1,371 @@
+				boutput(user, "<span class='notice'>[amtload] ores loaded from [O]!</span>")
+				boutput(user, "<span class='alert'>[src] rejects [rejected] anomalous ore(s).</span>")
+			else boutput(user, "<span class='alert'>No ore loaded!</span>")
+
+		else if (isitem(O))
+			quickload(user,O)
+		else
+			..()
+
+		src.updateUsrDialog()
+
+	proc/quickload(var/mob/living/user,var/obj/item/O)
+		if (!user || !O)
+			return
+		if(istype(O,/obj/item/raw_material/miracle) && !(O.material.name == "miraclium"))
+			boutput(user, "<span class='notice'>[src] rejects the anomalous ore.</span>")
+		user.visible_message("<span class='notice'>[user] begins quickly stuffing [O] into [src]!</span>")
+		var/staystill = user.loc
+		for(var/obj/item/M in view(1,user))
+			if (!M)
+				continue
+			if (M.type != O.type)
+				continue
+			if (!istype(M,/obj/item/raw_material/))
+				continue
+			if(!isSameMaterial(O.material,M.material))
+				continue
+			if(istype(M,/obj/item/raw_material/miracle) && !(M.material.name == "miraclium"))
+				continue // I eventually want to resolve this in a better way, but I legitimatly can't think of one right now - Urs
+			if (O.loc == user)
+				continue
+			if (O in user.contents)
+				continue
+			src.load_item(M)
+			M.set_loc(src)
+			playsound(get_turf(src), sound_load, 40, 1)
+			sleep(0.5)
+			if (user.loc != staystill) break
+		boutput(user, "<span class='notice'>You finish stuffing [O] into [src]!</span>")
+		return
+
+	attackby(obj/item/W as obj, mob/user as mob)
+		if (istype(W, /obj/item/raw_material/) && src.accept_loading(user))
+			if(istype(W,/obj/item/raw_material/miracle) && !(W.material.name == "miraclium"))
+				boutput(user, "<span class='alert'>[src] rejects the anomalous ore.</span>")
+				return // I eventually want to resolve this in a better way, but I legitimatly can't think of one right now - Urs
+			user.visible_message("<span class='notice'>[user] loads [W] into the [src].</span>", "<span class='notice'>You load [W] into the [src].</span>")
+			src.load_item(W,user)
+
+		else
+			src.health = max(src.health-W.force,0)
+			src.check_health()
+			..()
+
+	proc/check_health()
+		if(!src.health && !broken)
+			src.broken = 1
+			src.visible_message("<span class='alert'>[src] breaks!</span>")
+			src.icon_state = "ore_storage_unit-broken"
+
+	proc/load_item(var/obj/item/raw_material/R,var/mob/living/user)
+		if (!R)
+			return
+		var/amount_loaded = 0
+		if(R.amount == 1)
+			R.set_loc(src)
+			amount_loaded++
+			if (user && R)
+				user.u_equip(R)
+				R.dropped()
+		else if(R.amount>1)
+			R.set_loc(src)
+			for(R.amount,R.amount > 0, R.amount--)
+				var/obj/item/raw_material/new_mat = unpool(R.type)
+				new_mat.loc = src
+				amount_loaded++
+			if (user && R)
+				user.u_equip(R)
+				R.dropped()
+			pool(R)
+		update_ore_amount(R.material_name,amount_loaded)
+
+
+	proc/accept_loading(var/mob/user,var/allow_silicon = 0)
+		if (!user)
+			return 0
+		if (src.status & BROKEN || src.status & NOPOWER)
```

Comment:
you use a flag for broken here but when it breaks you set `var/broken` to 1 and don't adjust status. Am I missing something?

Reply:
one is the machinery flag and the other is the flag I coded myself. I didn't know what to do with the base level machinery flag 'cause i didn't know if it was used by parent for something else

Reply:
Double checked, and it isn't the only machinery to do this. I'm not sure what's up with it, but I'm calling this resolved for now since I don't know if the BROKEN flag is actually set by anything else and don't want to mess with it.